fe2-android/app/src/test/ui/robolectric/com/github/dedis/popstellar/ui/lao/witness/WitnessingFragmentTest.java

```diff
@@ -73,4 +84,51 @@ public void testBackButtonBehaviour() {
     // Check current fragment displayed is event list
     getEventListFragment().check(matches(isDisplayed()));
   }
+
+  @Test
+  public void testTabMenu() {
+    witnessingTabs().check(matches(isDisplayed()));
+
+    // Check that by default the witnesses are displayed
+    witnessMessageFragment().check(doesNotExist());
+    witnessesFragment().check(matches(isDisplayed()));
+
+    // Select the messages tab
+    witnessingTabs().perform(selectTabAtPosition(1));
+
+    // Check that the witness messages are displayed and the witnesses are not displayed
+    witnessMessageFragment().check(matches(isDisplayed()));
+    witnessesFragment().check(doesNotExist());
+
+    // Select the witnesses tab
+    witnessingTabs().perform(selectTabAtPosition(0));
+
+    // Check that the witnesses are displayed and the witnesses messages are not displayed
+    witnessMessageFragment().check(doesNotExist());
+    witnessesFragment().check(matches(isDisplayed()));
+  }
+
+  private static ViewAction selectTabAtPosition(final int tabIndex) {
+    return new ViewAction() {
+      @Override
+      public Matcher<View> getConstraints() {
+        return isAssignableFrom(TabLayout.class);
+      }
+
+      @Override
+      public String getDescription() {
+        return "select tab at index " + tabIndex;
+      }
+
+      @Override
+      public void perform(UiController uiController, View view) {
+        TabLayout tabLayout = (TabLayout) view;
+        TabLayout.Tab tab = tabLayout.getTabAt(tabIndex);
+        if (tab != null) {
+          tab.select();
+        }
+      }
+    };
+  }
+
```

Comment:
Can you transfer that to the utils class for the matcher (MatcherUtils) or to the PageObject, so it's more reusable also in the future?